.github/workflows/buf-pull-request.yml

```diff
@@ -69,6 +69,9 @@ jobs:
         run: |
           cd tools/proto-compiler
           cargo run
+          cd ../..
+          # Ignore changes to the unstable proto_descriptor output
+          git checkout -- proto/src/gen/proto_descriptor.bin
```

Comment:
@conorsch does this look OK to you? I asked in the Tokio discord and was told that the descriptor output depends on the specific protoc and so might not be stable / deterministic.  My thinking is that since it's generated by the tool, in practice there's no situation where we'd need to catch a change that wouldn't be caught by changes to the Rust codegen.

Reply:
Yes, this makes sense! There's actually a seldom-encountered race where the buf lock file can change due to an out of band (i.e. not a dep we control) version bump. I'll file an issue to track that, because I suspect the upcoming rewrite of the proto-compiler tool might fix it, too.

Reply:
> There's actually a seldom-encountered race where the buf lock file can change due to an out of band (i.e. not a dep we control) version bump. I'll file an issue to track that

Ahoy: #2321